task/sensor/src/main.rs

```diff
@@ -51,25 +57,82 @@ impl idl::InOrderSensorImpl for ServerImpl {
 
     fn get_reading(
         &mut self,
-        _: &RecvMessage,
+        msg: &RecvMessage,
         id: SensorId,
     ) -> Result<Reading, RequestError<SensorError>> {
+        self.get_raw_reading(msg, id)
+            .map_err(|e| e.map_runtime(SensorError::from))
+            .and_then(|(value, timestamp)| match value {
+                Ok(value) => Ok(Reading { value, timestamp }),
+                Err(e) => Err(<SensorError as From<_>>::from(e).into()),
```

Comment:
Nit - can this be

```suggestion
                Err(e) => Err(SensorError::from(e).into()),
```

Reply:
yes, thank you (I disliked how this looked before)!